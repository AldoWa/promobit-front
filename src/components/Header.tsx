import { Box, Image } from '@chakra-ui/react'
import Logo from '../assets/logo.svg';
import { Container } from '@chakra-ui/react';


const Header: React.FC = () => {
  return (
    <>
      <Box bg='purple.700' py={4}>
        <Container maxW='1214px'>
          <Image src={Logo}
            aria-label='Logo'
            alt='Logo'>
          </Image>
        </Container>
      </Box>
    </>
  );
};

export default Header;
