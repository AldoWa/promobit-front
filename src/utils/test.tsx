import { RenderOptions, render } from '@testing-library/react'

import { ChakraProvider } from '@chakra-ui/react'

interface AllTheProvidersProps {
  children: React.ReactNode
}

const AllTheProviders = ({children}: AllTheProvidersProps) => {
  return (
    <ChakraProvider>
      {children}
    </ChakraProvider>
  )
}

const customRender = (ui:React.ReactElement, options?: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// eslint-disable-next-line react-refresh/only-export-components
export * from '@testing-library/react'

// override render method
export {customRender as render}