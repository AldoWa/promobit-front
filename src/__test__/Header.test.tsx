import { render, screen } from "../utils/test"
import Header from "../components/Header";

describe('Header', () => {
  test("Should render the header", () => {
    render(<Header />)
    const element = screen.getByTestId('header-box')
    expect(element).toBeInTheDocument()
    expect(element).toBeVisible()
})

  test("Should match snapshot", () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot();
  })
})