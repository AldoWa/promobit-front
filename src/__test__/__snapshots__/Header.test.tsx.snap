// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
<div>
  <div
    class="css-qbb6fp"
    data-testid="header-box"
  >
    <div
      class="chakra-container css-1kw8fss"
      data-testid="header-container"
    >
      <img
        alt="Logo"
        aria-label="Logo"
        class="chakra-image css-0"
        data-testid="header-logo"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
